map2loop/sample_storage.py

```diff
@@ -0,0 +1,268 @@
+from abc import ABC, abstractmethod
+from .m2l_enums import Datatype, SampleType, StateType
+from .sampler import SamplerDecimator, SamplerSpacing, Sampler
+import beartype
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .project import Project
+
+
+class AccessStorage(ABC):
```

Comment:
Is it really necessary to have an abstract class for this? I can't think of a case where there are multiple sample supervisors